core/controllers/profile_test.py

```diff
@@ -964,6 +964,29 @@ def test_get_delete_account_page(self):
         self.assertIn(b'<oppia-root></oppia-root>', response.body)
 
 
+class SubscribeToAndroidListTests(test_utils.GenericTestBase):
+
+    def test_put_function(self):
+        # The GET function should not throw any error and should return status
+        # 200. No other check required here.
+        swap_add_fn = self.swap(
+            user_services, 'add_user_to_android_list', lambda *args: True)
+
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.login(self.VIEWER_EMAIL)
+
+        csrf_token = self.get_new_csrf_token()
+        with swap_add_fn:
+            json_response = self.put_json(
+                '/subscribetoandroidlist', {
+                    'email': 'email@example.com',
+                    'name': 'Name'
+                }, csrf_token=csrf_token)
+            self.assertEqual(json_response, {'status': True})
```

Comment:
I think you should test validation for error cases (bad emails, incomplete requests, etc.) -- make sure a 404 or similar happens.

Reply:
Done.